src/wmtk/operations/polygon_mesh/Splice.cpp

```diff
@@ -0,0 +1,156 @@
+#include "Splice.hpp"
+
+
+namespace {
+// Implicit opposite map for halfedges paired as [e] = {2*e, 2*e + 1}
+long implicit_opp(long h)
+{
+    return ((h % 2) == 0) ? (h + 1) : (h - 1);
+}
+} // namespace
+
+namespace wmtk::operations::polygon_mesh {
+
+Splice::Splice(Mesh& m, const Tuple& h, const Tuple& g, const OperationSettings<Splice>& settings)
+    : AtomicOperation(m)
+    , m_first_tuple(h)
+    , m_second_tuple(g)
+    , m_settings(settings)
+{}
+
+std::string Splice::name() const
+{
+    return "polygon_mesh_splice";
+}
+
+bool Splice::execute()
+{
+    assert(precondition());
+
+    // Old halfedge indices
+    long old_h_id = get_halfedge_from_tuple(m_first_tuple);
+    long old_g_id = get_halfedge_from_tuple(m_second_tuple);
+    long old_hn_id = get_next(old_h_id);
+    long old_gn_id = get_next(old_g_id);
+
+    // Do nothing for identical halfedges
+    if (old_hn_id == old_gn_id) {
+        return true;
+    }
+
+    // Old face indices
+    long old_hf_id = get_face(old_h_id);
```

Comment:
there's generally a lot of code in here that is in pairs - i'd prefer if you doubled up everything / used ranged for loops or std::transform to implement things. I definitely can't keep track of if you used `g` / `h` consistently everywhere. Alternatively you can isolate teh parts that only manipulate h or g with other functions to invoke chunks of things once rather than have so much duplicated code

Reply:
I Added size 2 arrays and used ranged for loops where possible. std::transform does not work well with non static member functions, so I kept some code duplication in the initialization of the next, face, and vertex arrays.

Reply:
is this due to lambda capture stuff?